File: ExampleApp/app/src/main/java/com/motiv/example/UsersFragmentPresenter.java
package com.motiv.example;

import com.motiv.example.dao.DaoRepository;
import com.motiv.example.dao.LocalStorage;

public class UsersFragmentPresenter implements UsersFragmentContract.Presenter {

    private UsersFragmentContract.View view;
    private GoApi goApi;
    private AuthApi authApi;
    private DaoRepository daoRepository;
    private LocalStorage localStorage;

    public UsersFragmentPresenter(
            UsersFragmentContract.View view,
            GoApi goApi,
            AuthApi authApi,
            DaoRepository daoRepository,
            LocalStorage localStorage) {
        this.view = view;
        this.goApi = goApi;
        this.authApi = authApi;
        this.daoRepository = daoRepository;
        this.localStorage = localStorage;
    }

    @Override
    public void goApigetUsersList() {
        goApi.getUsersList(
                new com.motiv.example.OnResponseListener<com.motiv.example.UsersResponse>() {
                    @Override
                    public void onSuccess(com.motiv.example.UsersResponse argument0) {
                        view.usersListAdaptersetData(argument0.getResult());
                        view.showToast(argument0.getMeta().getMessage());
                    }

                    @Override
                    public void onError(Exception argument0) {
                        view.showToast("error fetching data");
                    }
                });
    }

    @Override
    public void eloonItemClick(int argument0, com.motiv.example.User argument1) {
        view.navigationControllerstartUserActivity(argument1);
    }
}

File: ExampleApp/app/src/main/java/com/motiv/example/UsersFragmentPresenter.kt
 
package com.motiv.example
import androidx.fragment.app.*
import com.motiv.example.dao.DaoRepository
import com.motiv.example.dao.LocalStorage
import dagger.*
import dagger.android.*
import dagger.android.support.*
import javax.inject.*

public class UsersFragmentPresenter(val view: UsersFragmentContract.View, val goApi: GoApi, val authApi: AuthApi, val daoRepository: DaoRepository, val localStorage: LocalStorage) : UsersFragmentContract.Presenter {

    override fun goApigetUsersList() {
        goApi.getUsersList(object : com.motiv.example.OnResponseListener<com.motiv.example.UsersResponse> {
            override fun onSuccess(argument0: com.motiv.example.UsersResponse) {
                view.usersListAdaptersetData(argument0.getResult())
                view.showToast(argument0.getMeta().getMessage())
            } override fun onError(argument0: Exception) {
                view.showToast("error fetching data")
            } 
        })
    } override fun eloonItemClick(argument0: Int, argument1: com.motiv.example.User) {
        view.navigationControllerstartUserActivity(argument1)
    }
}
